benchmarks/src/fs2/data/benchmarks/CsvParserBenchmarks.scala

```diff
@@ -16,19 +16,19 @@ import cats.effect.SyncIO
 @Measurement(iterations = 5, time = 2)
 class CsvParserBenchmarks {
 
-  var csvContent: List[List[Char]] = _
+  var csvContent: List[String] = _
 
   @Setup
   def readCsv(): Unit = csvContent = {
-    File("../../../../../benchmarks/resources/benchmark.csv").contentAsString.grouped(4096).toList.map(_.toList)
+    File("../../../../../benchmarks/resources/benchmark.csv").contentAsString.grouped(4096).toList
   }
 
-  def csvStream: Stream[SyncIO, Char] = Stream.emits(csvContent).flatMap(Stream.emits).covary[SyncIO]
+  def csvStream: Stream[SyncIO, String] = Stream.emits(csvContent).covary[SyncIO]
 
   @Benchmark
   def parseRows(): Unit = {
     csvStream
-      .through(fs2.data.csv.rows[SyncIO](','))
+      .through(fs2.data.csv.rows[SyncIO, String](','))
```

Comment:
Have you run this benchmark to see the difference before and after this PR? I love the new abstraction, but given that it so critical for the performance of the whole library, IMHO we should run (and keep) some benchmarks to prove this is worth it. Maybe a simple benchmark comparing dumping `Char`s into a `Blackhole` from the old `Stream.flatMap(Stream.emits(_))` vs the new `CharLikeChunks` on a stream of `String`s?

Reply:
Yes I did, with the code you added, here are the results:

```
# master

Benchmark                              Mode  Cnt      Score     Error  Units
CsvParserBenchmarks.parseRows          avgt    5  11047,991 ± 118,289  us/op
CsvParserBenchmarks.parseRowsBaseline  avgt    5   6290,640 ±  28,784  us/op

# charlikechunks

Benchmark                              Mode  Cnt     Score    Error  Units
CsvParserBenchmarks.parseRows          avgt    5  4239,121 ± 38,023  us/op
CsvParserBenchmarks.parseRowsBaseline  avgt    5     2,877 ±  0,025  us/op
```

The difference is huge, but since I also refactored the CSV parser to emit non-1 chunks, it might be explained by a combination of both things. I will add benchmarks for JSON and CSV, since they didn't have this big refactoring

Reply:
That's roughly a ~2.5 factor 😳 IMHO too big to ignore, so excited on how it performs on XML and JSON.